Build-FFmpeg.ps1

```diff
@@ -0,0 +1,137 @@
+param(
+
+    [ValidateSet('x86', 'x64', 'ARM', 'ARM64')]
+    [string[]] $Platforms = ('x86', 'x64', 'ARM', 'ARM64'),
+
+    <#
+        Example values:
+        14.1
+        14.2
+        14.16
+        14.16.27023
+        14.23.27820
+    #>
+    [version] $VcVersion = '14.1',
+
+    <#
+        Example values:
+        8.1
+        10.0.15063.0
+        10.0.17763.0
+        10.0.18362.0
+    #>
+    [version] $WindowsTargetPlatformVersion = '10.0.15063.0',
+
+    [ValidateSet('Debug', 'Release')]
+    [string] $Configuration = 'Release',
+
+    [System.IO.DirectoryInfo] $VSInstallerFolder = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer",
+
+    [System.IO.FileInfo] $Msys2Bin = 'C:\msys64\usr\bin\bash.exe'
+)
+
+[System.IO.DirectoryInfo] $vsLatestPath = & "$VSInstallerFolder\vswhere.exe" -latest -property installationPath -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
+
+Write-Host "Visual Studio Installation folder: [$vsLatestPath]"
+
+# Export full current PATH from environment into MSYS2
+$env:MSYS2_PATH_TYPE = 'inherit'
+
+if (! (Test-Path $PSScriptRoot\ffmpeg\configure)) {
+    Write-Error 'configure is not found in ffmpeg folder. Ensure this folder is populated with ffmpeg snapshot'
+    Exit
+}
+
+# 14.16.27023 => v141
+$platformToolSet = "v$($VcVersion.Major)$("$VcVersion.Minor"[0])"
+
+# [$env:PROCESSOR_ARCHITECTURE][$platform]
+$vcvarsArchs = @{
+    'x86' = @{
+        'x86'   = 'x86'
+        'x64'   = 'x86_amd64'
+        'ARM'   = 'x86_arm'
+        'ARM64' = 'x86_arm64'
+    }
+
+    'AMD64' = @{
+        'x86'   = 'amd64_x86'
+        'x64'   = 'amd64'
+        'ARM'   = 'amd64_arm'
+        'ARM64' = 'amd64_arm64'
+    }
+}
+
+$start = Get-Date
+
+foreach ($platform in $Platforms) {
+
+    Write-Host "Building FFmpeg for Windows 10 apps ${platform}..."
+
+    # Load environment from VCVARS.
+    $vcvarsArch = $vcvarsArchs[$env:PROCESSOR_ARCHITECTURE][$platform]
+    CMD /c "`"$vsLatestPath\VC\Auxiliary\Build\vcvarsall.bat`" $vcvarsArch uwp $WindowsTargetPlatformVersion && SET" | . {
+        PROCESS {
+            if ($_ -match '^([^=]+)=(.*)') {
+                if ($matches[1] -notin 'HOME') {
+                    Set-Item -Path "Env:\$($Matches[1])" -Value $Matches[2]
+                }
+            }
+        }
+    }
+
+    New-Item -ItemType Directory -Force $PSScriptRoot\Libs\Build\$platform -OutVariable libs
+
+    ('lib', 'licenses', 'include', 'build') | ForEach-Object {
+        New-Item -ItemType Directory -Force $libs\$_
+    }
+    # Clean platform-specific build dir.
+    Remove-Item -Force -Recurse $libs\build\*
+
+    MSBuild.exe .\Libs\zlib\SMP\libzlib.vcxproj `

```

Comment:
Certainly. I'll implement error handling at the same level as the current BAT file.
Didn't add it initially because I'm looking for a nicer syntax than a bunch of try-catch blocks.

Reply:
Will do. This is pending...